components/callbacks/fx.go

```diff
@@ -60,23 +67,20 @@ func HTTPCallerProviderProvider(
 			}
 
 			callbackSource := r.Header.Get(callbackSourceHeader)
-			for clusterName, clusterInfo := range clusterMetadata.GetAllClusterInfo() {
+			for _, clusterInfo := range clusterMetadata.GetAllClusterInfo() {
 				if callbackSource == clusterInfo.ClusterID {
-					internalClient, err := httpClientCache.Get(clusterName)
-					if err != nil {
-						logger.Warn("HTTPCallerProviderProvider unable to get FrontendHTTPClient for callback target cluster. Using default HTTP client.", tag.SourceCluster(clusterMetadata.GetCurrentClusterName()), tag.TargetCluster(clusterName))
-						return client.Do(r)
-					}
-					r.URL.Scheme = internalClient.Scheme()
-					r.URL.Host = internalClient.Host()
-					r.Host = internalClient.Host()
+					r.URL.Scheme = internalClient.Scheme
+					r.URL.Host = internalClient.Address
```

Comment:
Won't this value get overwritten by the internal client's RoundTripper?

Reply:
Only if it's resolved to via membership.